src/platform/datapath_winkernel.c

```diff
@@ -2923,6 +2973,20 @@ SocketSend(
         *(PINT)WSA_CMSG_DATA(CMsg) = SendData->ECN;
     }
 
+    if (Binding->Datapath->Features & CXPLAT_DATAPATH_FEATURE_SEND_DSCP) {
+        CMsg = (PWSACMSGHDR)&CMsgBuffer[CMsgLen];
+        CMsgLen += WSA_CMSG_SPACE(sizeof(INT));
+        CMsg->cmsg_level =
+            Route->LocalAddress.si_family == QUIC_ADDRESS_FAMILY_INET ?
+                IPPROTO_IP : IPPROTO_IPV6;
+        CMsg->cmsg_type =
+            Route->LocalAddress.si_family == QUIC_ADDRESS_FAMILY_INET ?
+                IP_TOS : IPV6_TCLASS;
```

Comment:
Can we do the `if` check only once here?
```suggestion
        if (Route->LocalAddress.si_family == QUIC_ADDRESS_FAMILY_INET) {
            CMsg->cmsg_level = IPPROTO_IP;
            CMsg->cmsg_type = IP_TOS;
        } else { // INET6
            CMsg->cmsg_level = IPPROTO_IPV6;
            CMsg->cmsg_type = IPV6_TCLASS;
        }
```

Reply:
I'll clean it up for readability, but I'll mention the compiler is smart enough to optimize this to one comparison.